sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -215,9 +215,19 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
 
       val finalOptions = sessionOptions ++ extraOptions.toMap ++ pathsOption
       val dsOptions = new CaseInsensitiveStringMap(finalOptions.asJava)
-      val table = userSpecifiedSchema match {
-        case Some(schema) => provider.getTable(dsOptions, schema)
-        case _ => provider.getTable(dsOptions)
+      val table = provider match {
+        case hasCatalog: SupportsCatalogOptions =>
+          val ident = hasCatalog.extractIdentifier(dsOptions)
+          val catalog = CatalogV2Util.getTableProviderCatalog(
+            hasCatalog,
+            sparkSession.sessionState.catalogManager,
+            dsOptions)
+          catalog.loadTable(ident)
```

Comment:
shall we always call `TableProvider.getTable`?

Reply:
if we call `loadTable`, how do we handle user-specified schema?

Reply:
We don't, we simply ignore it. If the TableProvider SupportsCatalogOptions, then we will *always* load the table through the catalog, therefore we don't need user options or partitioning info

Reply:
then shall we fail or log a warning if schema is specified by users?

BTW `SupportsCatalogOptions` seems not a mixin as it doesn't need anything from `TableProvider`.

Reply:
I don't have a strong opinion on whether to warn, fail, or ignore when there is a user-specified schema. Warnings are almost always ignored, so I'd rather fail and highlight the problem to the user.